src/uct/cuda/cuda_ipc/cuda_ipc_md.h

```diff
@@ -52,27 +52,50 @@ typedef struct {
 } uct_cuda_ipc_memh_t;
 
 
+#if HAVE_CUDA_FABRIC
+typedef enum uct_cuda_ipc_key_handle {
+    UCT_CUDA_IPC_KEY_HANDLE_TYPE_ERROR = 0,
+    UCT_CUDA_IPC_KEY_HANDLE_TYPE_LEGACY,
+    UCT_CUDA_IPC_KEY_HANDLE_TYPE_VMM,
+    UCT_CUDA_IPC_KEY_HANDLE_TYPE_MEMPOOL
+} uct_cuda_ipc_key_handle_t;
+
+
+typedef struct uct_cuda_ipc_md_handle {
+    union {
+        CUipcMemHandle        legacy;        /* Legacy IPC handle */
+        CUmemFabricHandle     fabric_handle; /* VMM/Mallocasync export handle */
+    } handle;
+    CUmemPoolPtrExportData    ptr;
+    CUmemoryPool              pool;
+    uct_cuda_ipc_key_handle_t handle_type;
```

Comment:
I'd move handle_type to be the first field in the struct, since it decides the usage of the rest of the fields